File: Aigilas/Aigilas/Strategies/StraightLineStrategy.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Aigilas.Creatures;
using SPX.Entities;
using SPX.Core;

namespace Aigilas.Strategies
{
    class StraightLineStrategy:IStrategy
    {
        public StraightLineStrategy(ICreature parent, params int[] targetTypes)
            : base(parent,Strategy.StraightLine)
        {
            foreach (var targetType in targetTypes)
            {
                _targets.AddTargetTypes(targetType);
            }
        }

        public override void Act()
        {
            _parent.MoveIfPossible(0, 1);
            if(_parent.GetLocation().GridY >= GameManager.TileMapHeight-1)
            {
                _parent.SetInactive();
            }
        }
    }
}

File: proj/src/aigilas/strategies/impl/StraightLineStrategy.java
package aigilas.strategies.impl;

import aigilas.creatures.ICreature;
import aigilas.strategies.IStrategy;
import aigilas.strategies.Strategy;
import spx.bridge.ActorType;
import spx.core.Settings;

public class StraightLineStrategy extends IStrategy {
    public StraightLineStrategy(ICreature parent, ActorType... targetTypes)

    {
        super(parent, Strategy.StraightLine);

        for (ActorType targetType : targetTypes) {
            _targets.AddTargetTypes(targetType);
        }
    }

    @Override
    public void Act() {
        _parent.MoveIfPossible(0, 1);
        if (_parent.GetLocation().GridY >= Settings.Get().tileMapHeight - 1) {
            _parent.SetInactive();
        }
    }
}